Feature: Phalanx Check in Galaxyview & Teamview Status Page

kejith opened the pull request:
With this PR, a feature is added that allows the galaxy view and status page of Teamview to display the ranges of certain phalanxes.

Galaxy View
![image](https://user-images.githubusercontent.com/5557752/219900538-28447c09-00da-4de1-9a54-5132a12d7277.png)

Teamview Status Page
![image](https://user-images.githubusercontent.com/5557752/219900511-494f3ae8-bfd1-45f3-954d-e6717f9f53e2.png)

Future Implementations planned
- Tooltip for Teamview Status Page to see which Phalanxes have the System in Range
- Add a List of Phalanxes beneath the Status Page (like the List in Planets/Reports)

Because I was a little bit careless and worked on the same branch as the PR i  accidently added the DB migration functionality and the setup for Typescript to this PR. So, yeah. YAY

`npx knex migrate:latest` should be run to apply DB changes. Maybe someone can come up with a Solution to run this Command after a Devcontainer build/Docker Compose Build?

anoff commented:
> npx knew migrate:latest should be run to apply DB changes. Maybe someone can up with a Solution to run this Command after a Devcontainer build/Docker Compose Build?

Maybe `postStartCommand`, see https://containers.dev/implementors/json_reference/

kejith commented:
> > npx knew migrate:latest should be run to apply DB changes. Maybe someone can up with a Solution to run this Command after a Devcontainer build/Docker Compose Build?
> 
> Maybe `postStartCommand`, see https://containers.dev/implementors/json_reference/

Added the DB-Migration Process to docker build and run process. I also added an SQL-Query that will be run over the existing reports table to pre populate the phalanxes table.

So now this patch should be working by just pulling and opening the devcontainer or running docker compose.

anoff commented on userscript/features/galaxyview.js:
```diff
@@ -355,14 +355,48 @@ function modifyTable (data, modfiyFn) {
   addUploadSection()
   modifyTable({}, modifyAddRankFromPopup)
   const data = getVisibleSystem()
+  console.log({ data })
```
can we get rid of this or wrap it in a `if (teamviewDebug)`

kejith replied:
OH NO, shame on me. PR incoming.

anoff reviewed (changes requested):
what about dat log eh?

anoff commented:
THANKS FOR YOUR CONTRIBUTION THIS LOOKS NICE (NOW THAT YOU REMOVED THE LOG)
